share/completions/mono.fish

```diff
@@ -1,29 +1,161 @@
+  \'loop\tLoop related optimizations\'\\
+  \'peephole\tPeephole postpass\'\\
+  \'precomp\tPrecompile all methods before executing Main\'\\
+  \'sched\tInstruction scheduling\'\\
+  \'shared\tEmit per-domain code\'\\
+  \'sse2\tSSE2 instructions on x86 [arch-dependency]\'\\
+  \'tailc\tTail recursion and tail calls\'\\
+  \'transport\tSpecify the transport that the debugger will use to communicate\')'\
+  -f -d 'Use optimizations'
+
+complete -c mono -l response -d 'Use a response file'
+complete -c mono -l runtime -a '(mono --version)' -f -d 'Use Mono version'
+complete -c mono -l server -d 'Optimize the virtual machine to be better suited for server operations'
+complete -c mono -l use-map-jit -d 'Generate code using MAP_JIT on MacOS'
+complete -c mono -l verify-all \
+    -d 'Verify mscorlib and assemblies in the global assembly cache for valid IL, and all user code for IL verifiability'
+
+# Development options
+complete -c mono -l debug -a '(__complete_mono , \'casts\tProduce a detailed error when throwing a InvalidCastException\'\\
+  \'mdb-optimizations\tDisable some JIT optimizations which are usually only disabled when running inside the debugger\'\\
+  \'gdb\tGenerate and register debugging information with gdb\')'\
+  -f -d 'Use the debugging mode in the runtime'
+
+complete -c mono -l profile -f -d 'Use a profiler module with the given arguments'
+complete -c mono -l trace -f -d 'Show method names as they are invoked'
+complete -c mono -l no-x86-stack-align -d 'Don\'t align stack frames on the x86 architecture'
+complete -c mono -l jitmap -d 'Generate a JIT method map in a /tmp/perf-PID.map file'
+
+# JIT maintainer options
+complete -c mono -l bisect -f -d 'This flag is used by the automatic optimization bug bisector'
+complete -c mono -l break -x \
+    -d 'Insert a breakpoint before the method whose name is \'method\' (namespace.class:methodname)'
+complete -c mono -l breakonex -d 'Use a breakpoint on exceptions'
+complete -c mono -l compile -x -d 'Compile a method (namespace.name:methodname)'
+complete -c mono -l compile-all -d 'Compile all the methods in an assembly'
+complete -c mono -l graph \
+    -a 'cfg\t"Control Flow Graph (CFG)" dtree\t"Dominator Tree" code\t"CFG showing code" ssa\t"CFG showing code after SSA translation" optcode\t"CFG showing code after IR optimizations"' \
+    -x -d 'Generate a postscript file with a graph with the details about the specified method'
+complete -c mono -l ncompile \
+    -d 'Instruct the runtime on the number of times that the method(-s) specified by --compile/--compile-all to be compiled'
+complete -c mono -l stats \
+    -d 'Display information about the work done by the runtime during the execution of an application'
+complete -c mono -l wapi \
+    -a 'hps\t"Delete the global semaphore" semdel\t"List the currently used handles"' \
+    -x -d 'Perform maintenance of the process shared data'
+complete -c mono -s v -l verbose -d 'Show more messages'
```

Comment:
This commit fixes an error that was introduced by the previous commit. This back-and-forth adds noise and makes it harder to review. If you're motivated to clean that up, try [git rebase](https://git-rebase.io/).
There are several other commits that should be squashed (like  88d774019 "Fix all completions").

Reply:
I've squashed several commits.